codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-resolve-interceptor-with-multiauthsigv4a.java

```diff
@@ -0,0 +1,171 @@
+        DatabaseEndpointParams.Builder builder = DatabaseEndpointParams.builder();
+        builder.region(AwsEndpointProviderUtils.regionBuiltIn(executionAttributes));
+        builder.endpoint(AwsEndpointProviderUtils.endpointBuiltIn(executionAttributes));
+        setContextParams(builder, executionAttributes.getAttribute(AwsExecutionAttribute.OPERATION_NAME), request);
+        setStaticContextParams(builder, executionAttributes.getAttribute(AwsExecutionAttribute.OPERATION_NAME));
+        setOperationContextParams(builder, executionAttributes.getAttribute(AwsExecutionAttribute.OPERATION_NAME), request);
+        return builder.build();
+    }
+
+    private static void setContextParams(DatabaseEndpointParams.Builder params, String operationName, SdkRequest request) {
+    }
+
+    private static void setStaticContextParams(DatabaseEndpointParams.Builder params, String operationName) {
+    }
+
+    private <T extends Identity> SelectedAuthScheme<T> authSchemeWithEndpointSignerProperties(
+        List<EndpointAuthScheme> endpointAuthSchemes, SelectedAuthScheme<T> selectedAuthScheme) {
+        for (EndpointAuthScheme endpointAuthScheme : endpointAuthSchemes) {
+            if (!endpointAuthScheme.schemeId().equals(selectedAuthScheme.authSchemeOption().schemeId())) {
+                continue;
+            }
+            AuthSchemeOption.Builder option = selectedAuthScheme.authSchemeOption().toBuilder();
+            if (endpointAuthScheme instanceof SigV4AuthScheme) {
+                SigV4AuthScheme v4AuthScheme = (SigV4AuthScheme) endpointAuthScheme;
+                if (v4AuthScheme.isDisableDoubleEncodingSet()) {
+                    option.putSignerProperty(AwsV4HttpSigner.DOUBLE_URL_ENCODE, !v4AuthScheme.disableDoubleEncoding());
+                }
+                if (v4AuthScheme.signingRegion() != null) {
+                    option.putSignerProperty(AwsV4HttpSigner.REGION_NAME, v4AuthScheme.signingRegion());
+                }
+                if (v4AuthScheme.signingName() != null) {
+                    option.putSignerProperty(AwsV4HttpSigner.SERVICE_SIGNING_NAME, v4AuthScheme.signingName());
+                }
+                return new SelectedAuthScheme<>(selectedAuthScheme.identity(), selectedAuthScheme.signer(), option.build());
+            }
+            if (endpointAuthScheme instanceof SigV4aAuthScheme) {
+                SigV4aAuthScheme v4aAuthScheme = (SigV4aAuthScheme) endpointAuthScheme;
+                if (v4aAuthScheme.isDisableDoubleEncodingSet()) {
+                    option.putSignerProperty(AwsV4aHttpSigner.DOUBLE_URL_ENCODE, !v4aAuthScheme.disableDoubleEncoding());
+                }
+                if (!hasRegionSet(selectedAuthScheme) && v4aAuthScheme.signingRegionSet() != null) {
+                    RegionSet regionSet = RegionSet.create(v4aAuthScheme.signingRegionSet());
+                    option.putSignerProperty(AwsV4aHttpSigner.REGION_SET, regionSet);
+                }
```

Comment:
If I'm remembering the priorities right from the discussion with Alex, the endpoint provider's setting takes precedence over the auth scheme resolver's setting. This looks like we're favoring the auth scheme resolver. Am I misreading or misremembering?

Reply:
The table doesn't include the auth scheme resolver. In the discussion with Alex, when we added that, endpoint resolver came out higher than auth scheme resolver.

Reply:
Updated a comment